sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -292,46 +315,72 @@ interface ApigatewayIntegration {
     connectionId?: pulumi.Output<string>;
 }
 
-function createSwaggerSpec(
-        name: string,
-        opts: pulumi.ResourceOptions,
-        routes?: Route[],
-        staticRoutes?: StaticRoute[],
-        proxyRoutes?: ProxyRoute[]): SwaggerSpec {
-
+function createSwaggerSpec(name: string, opts: pulumi.ResourceOptions, routes: Routes): SwaggerSpec {
     // Set up the initial swagger spec.
     const swagger: SwaggerSpec = {
         swagger: "2.0",
         info: { title: name, version: "1.0" },
         paths: {},
         "x-amazon-apigateway-binary-media-types": ["*/*"],
+        // Map paths the user doesn't have access to as 404.
+        // http://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html
+        "x-amazon-apigateway-gateway-responses": {
+            "MISSING_AUTHENTICATION_TOKEN": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+            "ACCESS_DENIED": {
+                "statusCode": 404,
+                "responseTemplates": {
+                    "application/json": "{\"message\": \"404 Not found\" }",
+                },
+            },
+        },
     };
 
     // Now add all the routes to it.
-    addRoutesToSwaggerSpec(name, swagger, routes, opts);
-    addStaticRoutesToSwaggerSpec(name, swagger, staticRoutes, opts);
-    addProxyRoutesToSwaggerSpec(name, swagger, proxyRoutes, opts);
+    for (const path in routes) {
+        if (routes.hasOwnProperty(path)) {
+            const route = routes[path];
+
+            switch (route.kind) {
+                case "EventHandler":
+                    addEventHandlerRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                case "Static":
+                    addStaticRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                case "Proxy":
+                    addProxyRouteToSwaggerSpec(name, swagger, path, route, opts);
+                    continue;
+                default:
+                    const exhaustiveMatch: never = route;
+                    throw new Error('Non-exhaustive match for route');
```

Comment:
nifty TS trick so that if we add a new route kind, we'll be forced to update this switch.